packages/components/tag/src/tag-body.tsx

```diff
@@ -1,16 +1,40 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, {
+  MouseEvent,
+  KeyboardEvent,
+  ReactNode,
+  ElementType,
+} from 'react';
 import styled from '@emotion/styled';
-import { css, useTheme } from '@emotion/react';
+import type { Theme } from '@emotion/react';
+import type { LocationDescriptor } from 'history';
+import { css, useTheme, SerializedStyles } from '@emotion/react';
 import {
   customProperties as vars,
   designTokens,
 } from '@commercetools-uikit/design-system';
 import Text from '@commercetools-uikit/text';
 
-const Body = styled.div``;
+type TTagBodyProps = {
+  as?: ElementType;
+  to?: string | LocationDescriptor;
+  type: string;
+  onClick?: (
```

Comment:
There is a conflict when we are using `to` and `onClick`. Fixing one will cause other to throw an error.

- When we are using `MouseEvent<HTMLDivElement>`, `onClick` prop is throwing this error
`Type '((event: MouseEvent<HTMLDivElement, MouseEvent> | KeyboardEvent<HTMLDivElement>) => void) | undefined' is not assignable to type 'MouseEventHandler<HTMLButtonElement> | undefined'.
  Type '(event: MouseEvent<HTMLDivElement, MouseEvent> | KeyboardEvent<HTMLDivElement>) => void' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
    Types of parameters 'event' and 'event' are incompatible.
      Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'MouseEvent<HTMLDivElement, MouseEvent> | KeyboardEvent<HTMLDivElement>'.
        Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'MouseEvent<HTMLDivElement, MouseEvent>'.
          Property 'align' is missing in type 'HTMLButtonElement' but required in type 'HTMLDivElement'.ts(2322)
lib.dom.d.ts(6651, 5): 'align' is declared here.
index.d.ts(1457, 9): The expected type comes from property 'onClick' which is declared here on type 'IntrinsicAttributes & { theme?: Theme | undefined; as?: ElementType<any> | undefined; } & ClassAttributes<HTMLButtonElement> & ButtonHTMLAttributes<...> & { ...; }'` 

- When we are using `MouseEvent<HTMLButtonElement>`, `to` prop is throwing this error

`Type '{ children: Element; to: LocationDescriptor<unknown> | undefined; as: ElementType<any> | undefined; css: (false | SerializedStyles | SerializedStyles[] | undefined)[]; onClick: ((event: MouseEvent<...> | KeyboardEvent<...>) => void) | undefined; }' is not assignable to type 'IntrinsicAttributes & { theme?: Theme | undefined; as?: ElementType<any> | undefined; } & ClassAttributes<HTMLButtonElement> & ButtonHTMLAttributes<...> & { ...; }'.
  Property 'to' does not exist on type 'IntrinsicAttributes & { theme?: Theme | undefined; as?: ElementType<any> | undefined; } & ClassAttributes<HTMLButtonElement> & ButtonHTMLAttributes<...> & { ...; }'`